### PR TITLE
Make package no arch

### PR DESCRIPTION
**Pull request by BastianZim**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Using the virtual packages introduced in https://github.com/conda/conda/pull/10214, this PR makes the feedstock noarch whilst only making it available on Unix. Thich is necessary due to the dependency on a builtin module, as outlined in https://github.com/conda-forge/staged-recipes/pull/13861#issuecomment-776113290.

### CODE REVIEW

**Comment by BastianZim** [comment]
@conda-forge-admin, please rerender

**Comment by mfeurer** [comment]
Hey, I just wanted to note that the pynisher should work with OSX, we just don't have unit tests for that yet.

**Comment by BastianZim** [comment]
Hey, yes that's fine and pynisher still be available on macOS. What I'm doing is making this package [noarch](https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python). Noarch means that the package can, theoretically, be installed on any OS and under any python version (Although I'm restricting that to `>=3.6`). The advantage is here that we can support any underlying platform, even if conda-forge cannot compile the package for this specific platform.
I had previously not done that since pynisher cannot be installed under windows which cannot be specified with noarch but conda recently introduced a way to specify this even for noarch packages, which I hadn't known about before. 
The reason only Linux tests are being run now is that, since noarch packages are available on any platform, there is no need to test them on each platform anymore.

**Comment by BastianZim** [comment]
@jakirkham Hi John, sorry to be pinging you out of the blue but further to our conversation on https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/1192 I have now implemented `__unix`as I believe that his package fits the requirements and it seems to be working as intended (The original reason for arch is here: https://github.com/conda-forge/staged-recipes/pull/13861#issuecomment-776113290).
Since you're the expert, would you mind just verifying that this is the correct usage? If so, I'd merge this and also add this to the conda-forge docs, as suggested.

I'm also thinking about switching this to GitHub so that we can run the unit tests but it depends on scikit-learn which is quite significant. So the import tests wouldn't be able to catch this anymore. Do you have an opinion here?

**Comment by BastianZim** [comment]
Just gently bumping this again @jakirkham

**Comment by jakirkham** [comment]
cc @conda-forge/core (in case others have thoughts on this approach for `noarch` on OS restricted packages)

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,23 +11,20 @@ source:
...
-    - python
+    - python >=3.6
   run:
     - docutils >=0.3
     - psutil
-    - python
+    - python >=3.6
     - setuptools
-
+    - __unix
```
Comment (beckermr) [inline recipe/meta.yaml:27]: This virtual package won't be around on OSX or windows. Thus this build won't install on those systems. Is that intended? I could not tell from the comment thread.

Comment (beckermr) [reply recipe/meta.yaml:27]: In particular, this statement is not true

> Hey, yes that's fine and pynisher still be available on macOS.

Comment (isuruf) [reply recipe/meta.yaml:27]: ? `__unix` implies Linux and macOS

Comment (beckermr) [reply recipe/meta.yaml:27]: ack! I did not read that properly 🤦 

I read linux not unix. Sorry!

**Comment by BastianZim** [comment]
Just to summarise: Pynisher fulfils all `noarch ` requirements but will never be available on Windows since it uses `resource`. I initially used skip for this but wanted to switch to the new `__unix` style to keep it `noarch` otherwise.

**Comment by BastianZim** [comment]
Thanks for the review, everybody! Is there anything that should be changed or can I merge this? Just double-checking since this would be one of the first packages using this. @jakirkham @isuruf @beckermr

**Review by isuruf — APPROVED** [review]

**Comment by BastianZim** [comment]
Thanks, @isuruf!

**Comment by jakirkham** [comment]
Since it sounds like we are ok with this approach, we should probably document it so that others can pick this up as well.

@BastianZim, would you be interested in sending a PR to the docs ( https://github.com/conda-forge/conda-forge.github.io ) explaining this with a simple example? 🙂

**Comment by isuruf** [comment]
I don't think conda-forge is the correct place to document this. It should be `conda-build` docs and we can have a FAQ entry pointing to it.

**Comment by BastianZim** [comment]
@jakirkham Thanks for your help here!

Yes, of course, will do! I was just waiting for https://github.com/conda/conda/pull/10511 to get merged so that I can link to the conda-docs.

Just to check regarding the formatting. I would add it in the `noarch: python`section here: https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python but should it be a note, like the other elements or normal text (Or something else?).

**Comment by BastianZim** [comment]
@isuruf Conda now has the packages but no explanation. Let me know where you'd prefer that?

**Comment by BastianZim** [comment]
I'm happy to add it to both docs, just let me know what you prefer.

**Comment by jakirkham** [comment]
> I don't think conda-forge is the correct place to document this. It should be `conda-build` docs and we can have a FAQ entry pointing to it.

We already supply info like [this about CMake, etc.]( https://conda-forge.org/docs/maintainer/knowledge_base.html )

**Comment by isuruf** [comment]
> We already supply info like this about CMake, etc.

Mostly when we differ from conda-build docs. For eg: CMake instructions are different from conda-build docs where we don't recommend using those toolchain files.

**Pull request merged**